engine-comparison/dispatcher.sh

```diff
@@ -5,61 +5,98 @@
 . $(dirname $0)/../common.sh
 . ${SCRIPT_DIR}/common-harness.sh
 
+
 build_engine() {
-  # [[ ! -e ~/$FENGINE_CONFIGS_DIR/$1 ]] && echo "cant do" && exit 1
-  # All of this code will probably be in a separate script
 
   FENGINE_CONFIG=$1
-
-  echo "Building $FENGINE_CONFIG"
+  [[ ! -e $FENGINE_CONFIG ]] && echo \
+    "Error: build_engine function called for FENGINE_CONFIG=$FENGINE_CONFIG,\
+    but this file can't be found" && return
+  echo "Creating fuzzing engine: $FENGINE_CONFIG"
   FENGINE_NAME=$(basename $FENGINE_CONFIG)
-  ENGINE_DIR=$WORK/fuzz-engines/${FENGINE_NAME}
-  rm -rf $ENGINE_DIR
-  mkdir $ENGINE_DIR
-
-  # Fuzzing activities go here
+  FENGINE_DIR=$WORK/fengine-builds/${FENGINE_NAME}
+  rm -rf $FENGINE_DIR
+  mkdir $FENGINE_DIR
+
+  . $FENGINE_CONFIG
+  # Build either engine
+
+  if [[ $FUZZING_ENGINE == "libfuzzer" ]]; then
+
+    if [[ ! -d ${LIBFUZZER_SRC}/standalone ]]; then
+      echo "Checking out libFuzzer"
+      svn co http://llvm.org/svn/llvm-project/llvm/trunk/lib/Fuzzer
+    fi
+
+  elif [[ $FUZZING_ENGINE == "afl" ]]; then
+    # [[ ! -d $LIBFUZZER_SRC ]] && echo "Can't do AFL before libfuzzer" && break
+    if [[ ! -d ${LIBFUZZER_SRC}/afl ]]; then
+      mkdir -p ${LIBFUZZER_SRC}/afl
+      svn co http://llvm.org/svn/llvm-project/llvm/trunk/lib/Fuzzer/afl ${LIBFUZZER_SRC}/afl
+    fi
+    echo "Building a version of AFL"
+    cd $WORK/fengine-builds
+    wget http://lcamtuf.coredump.cx/afl/releases/afl-latest.tgz
+    mkdir $FENGINE_DIR
```

Comment:
Looks like you're creating FENGINE_DIR a second time here (first time on line 19).